src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/OverlappingEventSourceTests.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using Microsoft.Diagnostics.Monitoring.TestCommon.Options;
+using Microsoft.Diagnostics.Monitoring.TestCommon.Runners;
+using Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests.Fixtures;
+using Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests.HttpApi;
+using Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests.Runners;
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Diagnostics.Monitoring.WebApi.Models;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests
+{
+    [TargetFrameworkMonikerTrait(TargetFrameworkMonikerExtensions.CurrentTargetFrameworkMoniker)]
+    [Collection(DefaultCollectionFixture.Name)]
+    public class OverlappingEventSourceTests
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly ITestOutputHelper _outputHelper;
+
+        public OverlappingEventSourceTests(ITestOutputHelper outputHelper, ServiceProviderFixture serviceProviderFixture)
+        {
+            _httpClientFactory = serviceProviderFixture.ServiceProvider.GetService<IHttpClientFactory>();
+            _outputHelper = outputHelper;
+        }
+
+#if NET6_0_OR_GREATER
+        private const string DefaultRuleName = "FunctionalTestRule";
+        private readonly TimeSpan TraceDuration = Timeout.InfiniteTimeSpan;
+        private const string ClientBaseUrl = "http://localhost:10000";
+        private const string AppUrl = "http://+:10000";
+        private const string AspNetUrlsKey = "ASPNETCORE_Urls";
+
+        /// <summary>
+        /// Validates that an AspNetResponseStatus trigger will fire following an HTTP trace.
+        /// </summary>
+        [Theory]
+        [InlineData(DiagnosticPortConnectionMode.Listen)]
+        public Task OverlappingEventSourceTests_AspNetResponseStatusTest(DiagnosticPortConnectionMode mode)
+        {
+            const int ExpectedResponseCount = 2;
+
+            TemporaryDirectory tempDirectory = new(_outputHelper);
+            string ExpectedFilePath = Path.Combine(tempDirectory.FullName, "file.txt");
+            string ExpectedFileContent = Guid.NewGuid().ToString("N");
+            string[] urlPaths = new string[] { "", "/Privacy", "" };
+
+            Task ruleStartedTask = null;
+            return ScenarioRunner.SingleTarget(
+                _outputHelper,
+                _httpClientFactory,
+                mode,
+                TestAppScenarios.AspNetSlow.Name,
```

Comment:
nit: this scenario is getting lumped in with AspNetSlow, despite being for ResponseStatus. Should these be split out, or switched to a more appropriate name that applies to all of our AspNet* triggers.